src/main/scala/com/scality/clueso/SparkUtils.scala

```diff
@@ -0,0 +1,129 @@
+package com.scality.clueso
+
+import java.io.{File, IOException}
+import java.net.URI
+
+import com.scality.clueso.query.{MetadataQuery, MetadataQueryExecutor}
+import com.typesafe.config.ConfigFactory
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.hadoop.conf.{Configuration => HadoopConfig}
+import org.apache.hadoop.fs.{FileStatus, FileSystem, Path, PathFilter}
+import org.apache.spark.sql.functions.{col, lit}
+import org.apache.spark.sql.{Row, SparkSession}
+
+
+object SparkUtils extends LazyLogging {
+  def loadCluesoConfig(confFilePath: String) = {
+    val parsedConfig = ConfigFactory.parseFile(new File(confFilePath))
+    val _config = ConfigFactory.load(parsedConfig)
+
+    new CluesoConfig(_config)
+  }
+
+  def buildHadoopFs(config: CluesoConfig) = {
+    FileSystem.get(new URI(PathUtils.landingURI(config)), hadoopConfig(config))
+  }
+
+  def hadoopConfig(config: CluesoConfig): HadoopConfig = {
+    val c = new HadoopConfig()
+    c.set("fs.s3a.connection.ssl.enabled", config.s3SslEnabled)
+    c.set("fs.s3a.endpoint", config.s3Endpoint)
+    c.set("fs.s3a.impl", "org.apache.hadoop.fs.s3a.S3AFileSystem")
+    c.set("fs.s3a.access.key", config.s3AccessKey)
+    c.set("fs.s3a.secret.key", config.s3SecretKey)
+    c.set("fs.s3a.path.style.access", config.s3PathStyleAccess)
+    c.set("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
+
+    c
+  }
+
+  def buildSparkSession(config : CluesoConfig) = {
+    SparkSession
+      .builder
+      .config("spark.sql.parquet.cacheMetadata", "false")
+      .config("spark.hadoop.fs.s3a.fast.upload", "true")
+      .config("spark.hadoop.fs.s3a.buffer.dir", "./tmp")
+      .config("spark.hadoop.fs.s3a.connection.ssl.enabled", config.s3SslEnabled)
+      .config("spark.hadoop.fs.s3a.endpoint", config.s3Endpoint)
+      .config("spark.hadoop.fs.s3a.impl", "org.apache.hadoop.fs.s3a.S3AFileSystem")
+      .config("spark.hadoop.fs.s3a.access.key", config.s3AccessKey)
+      .config("spark.hadoop.fs.s3a.secret.key", config.s3SecretKey)
+      .config("spark.hadoop.fs.s3a.path.style.access", config.s3PathStyleAccess)
+      .config("spark.sql.streaming.metricsEnabled", "true")
```

Comment:
will this impact performance?

Reply:
Always does but it's not that much. Spark is already collecting most of the things to show in UI.